src/ser.rs

```diff
@@ -1002,12 +1002,38 @@ where
             .map_err(Error::io)
     }
 
-    fn serialize_f32(self, _value: f32) -> Result<()> {
-        Err(key_must_be_a_string())
+    fn serialize_f32(self, value: f32) -> Result<()> {
+        tri!(self
+            .ser
+            .formatter
+            .begin_string(&mut self.ser.writer)
+            .map_err(Error::io));
+        tri!(self
+            .ser
+            .formatter
+            .write_f32(&mut self.ser.writer, value)
+            .map_err(Error::io));
```

Comment:
This would either produce invalid JSON or nonsense, depending on the formatter, if the float isn't finite.

Reply:
I've now added checks to ensure those values are finite, and tests to ensure serialization with `NaN`, `inf`, and `-inf` keys returns an error.